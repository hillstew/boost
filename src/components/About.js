import React from 'react';

export const About = () => {
  return <p>a lot of stuff of about be the boost</p>;
};
