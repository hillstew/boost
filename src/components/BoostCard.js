import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

export const BoostCard = ({ img, fetchDogImage }) => {
  const url = 'https://random.dog/';
  return (
    <div className="BoostCard">
      <Link to={`/send-boost/${img}`}>
        <img src={url + img} width="300px" alt="dog" />
      </Link>
      <button className="BoostCard-button" onClick={() => fetchDogImage()} />
    </div>
  );
};

BoostCard.propTypes = {
  img: PropTypes.string,
  fetchDogImage: PropTypes.func
};
