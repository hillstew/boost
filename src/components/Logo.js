import React from 'react';
import logo from '../images/betheboost.png';

export const Logo = () => {
  return (
    <footer className="Logo-footer">
      <img src={logo} alt="be the boost logo" height="200px" />
    </footer>
  );
};
