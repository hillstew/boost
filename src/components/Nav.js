import React from 'react';
import { NavLink } from 'react-router-dom';

export const Nav = () => {
  return (
    <nav className="Nav">
      <NavLink to="/saved">Saved</NavLink>
      <hr />
      {/* <NavLink to="/sent">Sent</NavLink>
      <hr /> */}
      <NavLink to="/about">About</NavLink>
      <hr />
    </nav>
  );
};
