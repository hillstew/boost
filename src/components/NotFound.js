import React from 'react';

export const NotFound = () => {
  return <p>not found</p>;
};
