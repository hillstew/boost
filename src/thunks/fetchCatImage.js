import { setLoading, setError, setCatImage } from '../actions';

export const fetchCatImage = () => {
  return async (dispatch) => {
    try {
      dispatch(setLoading(true));
      const url = 'https://api.thecatapi.com/v1/images/search';
      const res = await fetch(url);
      if (!res.ok) {
        throw Error(res.statusText);
      }
      dispatch(setLoading(false));
      let catImages = await res.json();
      let image = catImages[0].url;
      image = image.split('/').pop();
      dispatch(setCatImage(image));
    }
    catch (error) {
      dispatch(setError(error.message));
    }
  };
};