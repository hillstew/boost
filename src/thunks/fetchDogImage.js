import { setLoading, setError, setDogImage } from '../actions';
import { randomIndex } from '../utils/helper';

export const fetchDogImage = () => {
  return async dispatch => {
    try {
      dispatch(setLoading(true));
      const url = 'https://random.dog/doggos';
      const res = await fetch(url);
      const dogImages = await res.json();
      const randomImage = dogImages[randomIndex(dogImages.length - 1)];
      dispatch(setDogImage(randomImage));
    } catch (error) {
      console.log(error);
    }
    dispatch(setLoading(false));
  };
};
