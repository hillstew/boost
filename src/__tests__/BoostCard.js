import React from 'react';
import { BoostCard } from '../components/BoostCard';
import { shallow } from 'enzyme';
import { fetchDogImage } from '../thunks/fetchDogImage';
jest.mock('../thunks/fetchDogImage');

describe('BoostCard', () => {
  let mockImg = 'hfdskfj.jpg';
  let mockFetchDog;
  let wrapper;

  beforeEach(() => {
    mockImg = 'hfdskfj.jpg';
    mockFetchDog = jest.fn();
    wrapper = shallow(<BoostCard img={mockImg} fetchDogImage={mockFetchDog} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it.skip('should call fetchDogImage onClick', () => {
    let wrapper = shallow(<BoostCard img={mockImg} fetchDogImage={mockFetchDog} />);
    // const spy = jest.spyOn(wrapper.instance(), 'fetchDogImage');
    // wrapper.forceUpdate();
    wrapper.find('BoostCard-button').simulate('click');
    expect(wrapper.instance().props.fetchDogImage).toHaveBeenCalled();
  });
});
