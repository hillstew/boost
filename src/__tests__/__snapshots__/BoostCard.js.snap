// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoostCard should match the snapshot 1`] = `
<div
  className="BoostCard"
>
  <Link
    replace={false}
    to="/send-boost/hfdskfj.jpg"
  >
    <img
      alt="dog"
      src="https://random.dog/hfdskfj.jpg"
      width="300px"
    />
  </Link>
  <button
    className="BoostCard-button"
    onClick={[Function]}
  />
</div>
`;
